packages/uicore/src/components/ModalDialog/ModalDialog.tsx

```diff
@@ -128,8 +165,12 @@ export const ModalDialog: FC<ModalDialogProps> = ({
         </aside>
       )}
 
-      <div className={css.body} data-testid="modaldialog-body" ref={bodyRef} onScroll={manageScrollIndicators}>
-        {children}
+      <div className={cx(css.body, bodyShadowClass)} data-testid="modaldialog-body" ref={bodyRef}>
+        <div className={css.bodyContent}>
+          <div ref={bodyTopEdgeRef} data-position="top" />
+          <div>{children}</div>
+          <div ref={bodyBottomEdgeRef} data-position="bottom" />
+        </div>
```

Comment:
I'm not a fan of so many `div`s personally. I think we can potentially remove the edge ref `div`s and instead track the start and end of the content. I think we could drop from 5 `div`s to 2 and maintain essentially the same approach.

Reply:
I'll try to reduce the `div`s when implementing the scroll shadows component.